[ONNX] Update instance_norm implementation and support training (#60538)

BowenBao opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #64581 [ONNX] Enable export of __xor_ (#64042)
* #64580 [ONNX] Fix inplace fill_ dtype export mismatch (#64233)
* #64579 [ONNX] Update doc and error message for indexing export (#64290)
* #64578 [ONNX] Fix remainder export (#64230)
* #64381 Add onnx test for batched_nms (#53175)
* #64379 [ONNX] Fix gather squeeze axis in constant folding (#63588)
* #64378 [ONNX] Fix cuda test case (#63597)
* #64377 [ONNX] Fix input sequence for pad op (#60554)
* #64376 Fix empty size constant creation (#63607)
* **#64375 [ONNX] Update instance_norm implementation and support training (#60538)**

* Update the instance_norm track_running_stats=True implementation and support the training mode
* Reference: https://github.com/pytorch/pytorch/blob/9baf75c86edc7f6cd1c04bf9f42d18bc0d05f504/aten/src/ATen/native/Normalization.cpp#L532
* Fix https://github.com/pytorch/pytorch/issues/53887

Co-authored-by: hwangdeyu <dejack953@outlook.com>

Differential Revision: [D30919605](https://our.internmc.facebook.com/intern/diff/D30919605)

hwangdeyu committed:
[ONNX] Update instance_norm implementation and support training (#60538)

* Update the instance_norm track_running_stats=True implementation and support the training mode
* Reference: https://github.com/pytorch/pytorch/blob/9baf75c86edc7f6cd1c04bf9f42d18bc0d05f504/aten/src/ATen/native/Normalization.cpp#L532
* Fix https://github.com/pytorch/pytorch/issues/53887

Co-authored-by: hwangdeyu <dejack953@outlook.com>

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Update instance_norm implementation and support training (#60538)"


* Update the instance_norm track_running_stats=True implementation and support the training mode
* Reference: https://github.com/pytorch/pytorch/blob/9baf75c86edc7f6cd1c04bf9f42d18bc0d05f504/aten/src/ATen/native/Normalization.cpp#L532
* Fix https://github.com/pytorch/pytorch/issues/53887

Co-authored-by: hwangdeyu <dejack953@outlook.com>

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Update instance_norm implementation and support training (#60538)"


* Update the instance_norm track_running_stats=True implementation and support the training mode
* Reference: https://github.com/pytorch/pytorch/blob/9baf75c86edc7f6cd1c04bf9f42d18bc0d05f504/aten/src/ATen/native/Normalization.cpp#L532
* Fix https://github.com/pytorch/pytorch/issues/53887

Co-authored-by: hwangdeyu <dejack953@outlook.com>

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Update instance_norm implementation and support training (#60538)"


* Update the instance_norm track_running_stats=True implementation and support the training mode
* Reference: https://github.com/pytorch/pytorch/blob/9baf75c86edc7f6cd1c04bf9f42d18bc0d05f504/aten/src/ATen/native/Normalization.cpp#L532
* Fix https://github.com/pytorch/pytorch/issues/53887

Co-authored-by: hwangdeyu <dejack953@outlook.com>

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Update instance_norm implementation and support training (#60538)"


* Update the instance_norm track_running_stats=True implementation and support the training mode
* Reference: https://github.com/pytorch/pytorch/blob/9baf75c86edc7f6cd1c04bf9f42d18bc0d05f504/aten/src/ATen/native/Normalization.cpp#L532
* Fix https://github.com/pytorch/pytorch/issues/53887

Co-authored-by: hwangdeyu <dejack953@outlook.com>

[ghstack-poisoned]

msaroufim commented:
@msaroufim has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30919605).

BowenBao committed:
Update on "[ONNX] Update instance_norm implementation and support training (#60538)"


* Update the instance_norm track_running_stats=True implementation and support the training mode
* Reference: https://github.com/pytorch/pytorch/blob/9baf75c86edc7f6cd1c04bf9f42d18bc0d05f504/aten/src/ATen/native/Normalization.cpp#L532
* Fix https://github.com/pytorch/pytorch/issues/53887

Co-authored-by: hwangdeyu <dejack953@outlook.com>

Differential Revision: [D30919605](https://our.internmc.facebook.com/intern/diff/D30919605)

[ghstack-poisoned]

msaroufim commented:
@msaroufim has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30919605).

BowenBao committed:
Update on "[ONNX] Update instance_norm implementation and support training (#60538)"


* Update the instance_norm track_running_stats=True implementation and support the training mode
* Reference: https://github.com/pytorch/pytorch/blob/9baf75c86edc7f6cd1c04bf9f42d18bc0d05f504/aten/src/ATen/native/Normalization.cpp#L532
* Fix https://github.com/pytorch/pytorch/issues/53887

Co-authored-by: hwangdeyu <dejack953@outlook.com>

Differential Revision: [D30919605](https://our.internmc.facebook.com/intern/diff/D30919605)

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Update instance_norm implementation and support training (#60538)"


* Update the instance_norm track_running_stats=True implementation and support the training mode
* Reference: https://github.com/pytorch/pytorch/blob/9baf75c86edc7f6cd1c04bf9f42d18bc0d05f504/aten/src/ATen/native/Normalization.cpp#L532
* Fix https://github.com/pytorch/pytorch/issues/53887

Co-authored-by: hwangdeyu <dejack953@outlook.com>

Differential Revision: [D30919605](https://our.internmc.facebook.com/intern/diff/D30919605)

[ghstack-poisoned]

malfet commented:
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30919605).

BowenBao committed:
Update on "[ONNX] Update instance_norm implementation and support training (#60538)"


* Update the instance_norm track_running_stats=True implementation and support the training mode
* Reference: https://github.com/pytorch/pytorch/blob/9baf75c86edc7f6cd1c04bf9f42d18bc0d05f504/aten/src/ATen/native/Normalization.cpp#L532
* Fix https://github.com/pytorch/pytorch/issues/53887

Co-authored-by: hwangdeyu <dejack953@outlook.com>

Differential Revision: [D30919605](https://our.internmc.facebook.com/intern/diff/D30919605)

[ghstack-poisoned]
